src/react/account/AccountRoleSelectModal.tsx

```diff
@@ -0,0 +1,167 @@
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useRouteMatch } from 'react-router';
+import Table from '@scality/core-ui/dist/components/tablev2/Tablev2.component';
+import { Button } from '@scality/core-ui/dist/next';
+import { Tooltip } from '@scality/core-ui';
+import { AppState } from '../../types/state';
+import { closeAccountRoleSelectModal } from '../actions';
+import { CustomModal as Modal, ModalBody } from '../ui-elements/Modal';
+import { regexArn, SCALITY_INTERNAL_ROLES, useAccounts } from '../utils/hooks';
+import {
+  useCurrentAccount,
+  useDataServiceRole,
+} from '../DataServiceRoleProvider';
+import { getRoleArnStored, setRoleArnStored } from '../utils/localStorage';
+import { Icon } from '../ui-elements/Help';
+
+function AccountRoleSelectModal() {
+  const dispatch = useDispatch();
+  const accounts = useAccounts();
+  const { path } = useRouteMatch();
+
+  const { account, selectAccountAndRoleRedirectTo } = useCurrentAccount();
+  const { roleArn } = useDataServiceRole();
+  const [assumedRoleArn, setAssumedRoleArn] = useState(roleArn);
+  const [currentAccountName, setCurrentAccountName] = useState(account?.Name);
+  const isModalOpen = useSelector(
+    (state: AppState) => state.uiAccounts.showAccountRoleSelect,
+  );
+
+  const accountsWithRoles = [];
+  accounts.forEach((account) => {
+    const accountName = account.Name;
+    account.Roles.forEach((role) => {
+      const rolePathName = regexArn.exec(role.Arn)?.groups['role_name'] || '';
+      const rolePath = rolePathName.includes('/')
+        ? rolePathName.split('/')[0]
+        : '/';
+      const roleName = rolePathName.split('/')[1] || rolePathName;
```

Comment:
I think a role path can include several sub paths. The role name would only be the last part of the arn (after the last /)

Reply:
This is confirmed by the example here https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html

Reply:
The regexArn could extract the path for you.

```
export const regexArn =
  /arn:aws:iam::(?<account_id>\d{12}):role\/(?<role_path>(?:[^\/]*\/)*)(?<role_name>[^\/]+)$/;
```